cpp/src/slice2cs/Gen.cpp

```diff
@@ -480,16 +480,7 @@ Slice::CsVisitor::writeDispatch(const InterfaceDefPtr& p)
                 }
                 else
                 {
-                    _out << nl << typeS << ' ' << param << ';';
-                    StructPtr st = dynamic_pointer_cast<Struct>((*pli)->type());
-                    if (st && isValueType(st))
-                    {
-                        _out << nl << param << " = new " << typeS << "();";
-                    }
-                    else if (st || isClass)
-                    {
-                        _out << nl << param << " = null;";
-                    }
+                    _out << nl << typeS << ' ' << param << " = default;";
```

Comment:
= default is always fine.

It's really to avoid warnings/errors from writeMarshalUnmarshalParams that should not read unassigned values.